Fixed #726 Mockito Public API Support

thekingn0thing opened the pull request:
Fixed #726 Mockito Public API Support

Incapability with PowerMock 1.x:

* Possible incapability due toString methods returns by default name of mocked.
* Fully drop supporting of Mockito 1.x 
* Removed deprecated `PowerMockito verifyStatic()`. It is replaced by `verifyStatic(Class)`
* `mockStatic` does not reset mocking process anymore. As result you may get the `UnfinishedVerificationException` or `UnfinishedStubbingException`

Other PowerMockito changes:
* Removed all Mockito copy-pasted code 
* Use delegate MockMaker to prevent issue with replacing Mockito MockMaker when PowerMock in class path (see #769, mockito/mockito#1004)
* Remove code regarding 
* Replace using Mockito internal classes with Mockito public API, except annotation processing (#838).

Commit:
Fixes #726 Mockito Public API Support
Using Mockito API for stubbing and creating invocation.

Commit:
Fixes #726 Mockito Public API Support
Replace using ThrowsException and other Mockito's internals answers by calling public API.

Commit:
Fixes #726 Mockito Public API Support
Using Mockito API for stubbing and creating invocation.

Commit:
Fixes #726 Mockito Public API Support
Replace using ThrowsException and other Mockito's internals answers by calling public API.

Commit:
Fixes #726 Mockito Public API Support
Remove deprecated `verifyStatic()`.

Commit:
Fixes #726 Mockito Public API Support
Remove reseting mocking process in PowerMockito

Commit:
Fixes #726 Mockito Public API Support
Remove unused Mockito related code.

Commit:
Fixes #726 Mockito Public API Support
Replace dirty approach for checking if call to real method was handled by mockito.

Commit:
Fixes #726 Mockito Public API Support
Remove using Mockito internal class in default.properties

Commit:
Fixes #726 Mockito Public API Support
Remove `MockingFrameworkReporterFactory`  because it does not make sense anymore.

Commit:
Fixes #726 Mockito Public API Support
Clearing code: fix todo and refactor tests,

Commit:
Fixed #726 Mockito Public API Support
Clearing code and fixing test.

TimvdLippe commented:
Really nice to see so much code being deleted, because Mockito can take care of it. Good job!

thekingn0thing commented:
Not all code has been deleted due to moving to public API. It also includes dropping supporting Mockito 1.x. As result the whole module has been deleted.

Commit:
Fixes #726 Mockito Public API Support
Upgrade Mockito to 2.10.0 - release with a new Mockito API.

Commit:
Fixes #726 Mockito Public API Support
Fix tests that are failed only on travis due different order of executing.

Commit:
Fixes #726 Mockito Public API Support
Delete unnecessary mockito-modules file, because there is no mockito modules any more.
